client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -98,18 +98,20 @@ private final class PoolManager[F[_], A <: Connection[F]](
     * @param callback The callback to complete with the NextConnection.
     */
   private def createConnection(key: RequestKey, callback: Callback[NextConnection]): F[Unit] =
-    if (numConnectionsCheckHolds(key)) {
-      incrConnection(key) *> F.start {
-        Async.shift(executionContext) *> builder(key).attempt.flatMap {
-          case Right(conn) =>
-            F.delay(callback(Right(NextConnection(conn, fresh = true))))
-          case Left(error) =>
-            disposeConnection(key, None) *> F.delay(callback(Left(error)))
-        }
-      }.void
-    } else {
-      addToWaitQueue(key, callback)
-    }
+    F.ifM(F.delay(numConnectionsCheckHolds(key)))(
```

Comment:
`numConnectionsCheckHolds` accesses a mutable map, but it does not mutate it. I don't think this requires `delay`.

Every call also within the `if` is thunked within delay as well.